modules/gdscript/language_server/gdscript_workspace.cpp

```diff
@@ -53,7 +53,7 @@ void GDScriptWorkspace::remove_cache_parser(const String &p_path) {
 	Map<String, ExtendGDScriptParser *>::Element *parser = parse_results.find(p_path);
 	Map<String, ExtendGDScriptParser *>::Element *script = scripts.find(p_path);
 	if (parser && script) {
-		if (script->get() && script->get() == script->get()) {
+		if (script->get() == parser->get()) {
```

Comment:
Maybe the whole block of lines 55-70 could be simplified further? @Geequlim 
```cpp
if (parser) {
	memdelete(parser->get());
	parse_results.erase(p_path);
}
if (script) {
	if (script->get()) { // May have been freed already if same as parser.
		memdelete(script->get());
	}
	scripts.erase(p_path);
}
```

Reply:
yeah looks better, On it

Reply:
```
if (parser && parser->get()) {
	memdelete(parser->get());
	parse_results.erase(p_path);
}
if (script && script->get()) {
	memdelete(script->get());
	scripts.erase(p_path);
}
```

Reply:
I don't know, the previous logic would remove the path from `parse_results` and `scripts` even if the `ExtendGDScriptParser` instance is null, at least in some branches.

Reply:
```c++
if (parser) {
	if (parser->get()) memdelete(parser->get());
	parse_results.erase(p_path);
}
if (script) {
	if (script->get()) memdelete(script->get());
	scripts.erase(p_path);
}
```